app/src/main/java/com/my/journeyplanner/framework/JourneyPlannerResultConverterFactory.kt

```diff
@@ -0,0 +1,113 @@
+                this,
+                JourneyPlannerResult.FromAndToDisambiguationOptions::class.java,
+                annotations
+            )
+
+        val notIdentifiedConverterFactory =
+            retrofit.nextResponseBodyConverter<JourneyPlannerResult.NotIdentified>(
+                this,
+                JourneyPlannerResult.NotIdentified::class.java,
+                annotations
+            )
+
+        val fromDisambiguationOptionsConverterFactory =
+            retrofit.nextResponseBodyConverter<JourneyPlannerResult.FromDisambiguationOptions>(
+                this,
+                JourneyPlannerResult.FromDisambiguationOptions::class.java,
+                annotations
+            )
+
+        val toDisambiguationOptionsConverterFactory =
+            retrofit.nextResponseBodyConverter<JourneyPlannerResult.ToDisambiguationOptions>(
+                this,
+                JourneyPlannerResult.ToDisambiguationOptions::class.java,
+                annotations
+            )
+
+        return Converter { body ->
+            val bodyString = body.string()
+            val jsonBody = JSONObject(bodyString)
+            val hasFromLocationDisambiguation = jsonBody.has(FROM_LOCATION_DISAMBIGUATION)
+            val hasToLocationDisambiguation = jsonBody.has(TO_LOCATION_DISAMBIGUATION)
+            val hasFromLocationDisambiguationMatchStatus =
+                jsonBody.hasLocationDisambiguationKey(
+                    hasFromLocationDisambiguation,
+                    FROM_LOCATION_DISAMBIGUATION,
+                    MATCH_STATUS
+                )
+            val hasToLocationDisambiguationMatchStatus =
+                jsonBody.hasLocationDisambiguationKey(
+                    hasToLocationDisambiguation,
+                    TO_LOCATION_DISAMBIGUATION,
+                    MATCH_STATUS
+                )
+            val isNotIdentified =
+                hasFromLocationDisambiguationMatchStatus && hasToLocationDisambiguationMatchStatus
+            val isItineraryResult = jsonBody.isItineraryResult()
+            val hasFromLocationDisambiguationOptions =
+                jsonBody.hasLocationDisambiguationKey(
+                    hasFromLocationDisambiguation,
+                    FROM_LOCATION_DISAMBIGUATION,
+                    DISAMBIGUATION_OPTIONS
+                )
+            val hasToLocationDisambiguationOptions =
+                jsonBody.hasLocationDisambiguationKey(
+                    hasToLocationDisambiguation,
+                    TO_LOCATION_DISAMBIGUATION,
+                    DISAMBIGUATION_OPTIONS
+                )
+            val isFromAndToLocationDisambiguationOptions =
+                hasFromLocationDisambiguationOptions && hasToLocationDisambiguationOptions
+            val isFromDisambiguationOptions =
+                hasFromLocationDisambiguationOptions && !hasToLocationDisambiguationOptions
+            val isToDisambiguationOptions =
```

Comment:
from what I gathered, there was a field called type which had a full qualified type. What happened to that and why did you decide to understand what the data was comprised of?

Reply:
ItineraryResult i.e. exact match
{"$type":"Tfl.Api.Presentation.Entities.JourneyPlanner.ItineraryResult, Tfl.Api.Presentation.Entities",...
notidentified i.e. no match either way
{"$type":"Tfl.Api.Presentation.Entities.JourneyPlanner.DisambiguationResult, Tfl.Api.Presentation.Entities",...
From To Disambiguation i.e. multiple options both ways
{"$type":"Tfl.Api.Presentation.Entities.JourneyPlanner.DisambiguationResult, Tfl.Api.Presentation.Entities",...
From DisambiguationResult i.e. mutliple options in from direction
{"$type":"Tfl.Api.Presentation.Entities.JourneyPlanner.DisambiguationResult, Tfl.Api.Presentation.Entities",...
To DisambiguationResult i.e. multiple options in to direction
{"$type":"Tfl.Api.Presentation.Entities.JourneyPlanner.DisambiguationResult, Tfl.Api.Presentation.Entities",...

The type is the same for the other types with the exception of ItineraryResult hence why I have to drill down a bit more

Reply:
Reduced data types from 5 to 2 and removed associated logic